refreshVersions/src/main/kotlin/de/fayard/versions/internal/UsedDependenciesWriting.kt

```diff
@@ -30,16 +39,41 @@ internal fun Project.writeUsedDependencies() {
 
     file.bufferedWriter().use { writer ->
         allDependencies.forEach { dependency ->
-            writer.append(dependency.group)
-            writer.append(':')
-            writer.append(dependency.name)
-            writer.append(':')
-            writer.append(dependency.version)
-            writer.appendln()
+            writer.appendDependency(
```

Comment:
One thing I want to change going forward is to do more in unit tests. 
Right now if I make changes in the plugin, I have no idea what will break in the projects you have at work. 
Not nice :)

So let have more unit tests, but how?

I'm a big believer in the "Functional core - Imperative shell" principle described here 

https://www.destroyallsoftware.com/screencasts/catalog/functional-core-imperative-shell

As it is, the function `writeUsedDependencies()` and `writer.appendDependency()` cannot be unit-tested easily. 

I would suggest to have a core function that takes a list of dependencies and return a string that corresponds to the file content (functional core) and unit test the hell out of it. 

Then the imperative shell will be just `file.writeText(dependencies.format())`

Reply:
As to the `appendDependency` function, it's an extension on `Appendable` so you can write a file-less unit test for it using a `StringBuilder` (like in `buildString { ... }`).

That said, I agree that testability can be improved, and that functional core, imperative shell is a goo approach.